force-app/main/default/classes/PMT_RefundService.cls

```diff
@@ -190,20 +224,103 @@ public inherited sharing class PMT_RefundService {
         return response;
     }
 
-    public void processDML() {
-        Savepoint sp = Database.setSavepoint();
-        try {
-            if (!this.refundRecords.isEmpty()) {
-                insert this.refundRecords;
+    public npe01__OppPayment__c buildRefundRecord(npe01__OppPayment__c originalPayment, Decimal refundAmount) {
+        refundAmount = (refundAmount == null) ? originalPayment.npe01__Payment_Amount__c : refundAmount; 
+        npe01__OppPayment__c refund = new npe01__OppPayment__c(
+            npe01__Payment_Amount__c = -1 * refundAmount,
+            npe01__Opportunity__c = originalPayment.npe01__Opportunity__c,
+            OriginalPayment__c = originalPayment.Id,
+            DebitType__c = (originalPayment.npe01__Payment_Amount__c == refundAmount)
+                ? PMT_RefundService.FULL_REFUND
+                : PMT_RefundService.PARTIAL_REFUND
+        );
+        
+        if (isMultiCurrencyEnabled) {
+            refund.put(UTIL_Currency.CURRENCY_ISO_CODE_FIELD, originalPayment.get(UTIL_Currency.CURRENCY_ISO_CODE_FIELD));
+        }
+
+        if (this.refundRecords == null) {
+            this.refundRecords = new List<npe01__OppPayment__c>();
```

Comment:
I don't love this pattern, but if this is the only place were initializing the List is needed, this should be fine.

Reply:
Initialize the list at class level now